CONTRIBUTING.md

```diff
@@ -1,57 +1,108 @@
-- Do not abuse `/*prettier-ignore*/`.
+- **Testing Requirement:**
+  - All changes must be thoroughly tested before submission. This includes:
+    - Unit tests.
+    - Integration tests.
+    - Manual checks, as necessary.
 
-## Using `/* prettier-ignore */`
+- **Repository History:**
+  - Keep the repository history clean and linear by using **rebase** and **squash** strategies effectively.
 
-### What is `/* prettier-ignore */`?
+---
 
-`/* prettier-ignore */` is a directive used in Prettier to prevent the code formatter from formatting a specific piece of code. This can be useful when you have lines of code that should remain unchanged for readability, debugging, or other specific reasons.
+## Pull Request Guidelines
 
-### When to Use `/* prettier-ignore */`
+1. **Draft Stage:**
+   - Begin each pull request (PR) in the `Draft` state to allow for incremental changes and early feedback.
 
-There are a few common scenarios where you might want to use `/* prettier-ignore */`:
+2. **Commit Best Practices:**
+   - Use **atomic commits**: Each commit should represent a single logical change.
+   - Commit frequently to track progress and facilitate collaboration.
 
-1. **Long Debug Lines**: When you have long lines of code for debugging purposes that you want to keep on a single line for easier readability.
-2. **Special Formatting**: When you have specific formatting that you want to maintain that Prettier would otherwise change.
-3. **Temporary Code**: When you have temporary code that is meant for testing and you don’t want it to be auto-formatted.
+3. **Rebase and Squash:**
+   - Before moving a PR out of the draft stage:
+     - Rebase your branch onto the latest `dev` branch.
+     - Squash commits to streamline the history. Multiple commits can remain if they represent logical steps.
 
-### How to Use `/* prettier-ignore */`
+4. **Code Compliance:**
+   - Ensure your code adheres to the project’s formatting and linting guidelines.
 
-To use `/* prettier-ignore */`, simply place the comment on the line before the code you want to ignore. Here is an example:
+5. **Testing:**
+   - Include test coverage and ensure your branch passes all required tests and CI/CD pipelines.
 
-#### Example
+6. **Branch Updates:**
+   - Regularly update your branch with the latest changes from `dev` using `--rebase` to avoid conflicts.
 
-```javascript
-/* prettier-ignore */
-if (ShardeumFlags.VerboseLogs) console.log('Lengths of blocks after pruning', Object.keys(blocksByHash).length, Object.keys(readableBlocks).length)
-```
+7. **Integration Guidelines:**
+   - Use **rebase** instead of merge to integrate changes from `dev` and maintain a clean history.
+   - Direct submissions to the `main` branch are **prohibited**. All changes must go through `dev` first.
+
+8. **PR Reviews:**
+   - Once your PR is ready, undraft it and request reviews.
+   - Be receptive to feedback and make necessary adjustments promptly.
+
+---
+
+## Standardized Formatting Guidelines
```

Comment:
under this section, bring back the prettier guidelines: https://github.com/shardeum/shardeum/blob/dev/CONTRIBUTING.md#using--prettier-ignore-, it is helpful for the new contributors.